compiletest does not say which test is problematic

this happened here  https://github.com/rust-lang/rust/pull/138853#issuecomment-2746122743 

```
2025-03-23T09:58:04.3107590Z    Compiling tracing-subscriber v0.3.18
2025-03-23T09:58:04.5068470Z [RUSTC-TIMING] regex_automata test:false 8.301
2025-03-23T09:58:04.5169020Z    Compiling rustfix v0.8.1
2025-03-23T09:58:04.7070410Z [RUSTC-TIMING] getopts test:false 0.626
2025-03-23T09:58:04.7171600Z [RUSTC-TIMING] anyhow test:false 0.499
2025-03-23T09:58:04.9296810Z [RUSTC-TIMING] rustfix test:false 0.414
2025-03-23T09:58:04.9397960Z    Compiling colored v2.2.0
2025-03-23T09:58:05.1137890Z [RUSTC-TIMING] semver test:false 0.405
2025-03-23T09:58:05.1143340Z    Compiling build_helper v0.1.0 (/Users/runner/work/rust/rust/src/build_helper)
2025-03-23T09:58:05.2949520Z [RUSTC-TIMING] colored test:false 0.357
2025-03-23T09:58:05.2956530Z    Compiling miropt-test-tools v0.1.0 (/Users/runner/work/rust/rust/src/tools/miropt-test-tools)
2025-03-23T09:58:05.6358490Z [RUSTC-TIMING] miropt_test_tools test:false 0.336
2025-03-23T09:58:05.6362710Z    Compiling home v0.5.11
2025-03-23T09:58:05.6926720Z [RUSTC-TIMING] home test:false 0.052
2025-03-23T09:58:05.6932500Z    Compiling glob v0.3.2
2025-03-23T09:58:06.2956820Z [RUSTC-TIMING] build_helper test:false 1.176
2025-03-23T09:58:06.2979930Z    Compiling compiletest v0.0.0 (/Users/runner/work/rust/rust/src/tools/compiletest)
2025-03-23T09:58:06.3486720Z [RUSTC-TIMING] glob test:false 0.652
2025-03-23T09:58:07.4381400Z [RUSTC-TIMING] tracing_subscriber test:false 3.127
2025-03-23T09:58:12.2066910Z [RUSTC-TIMING] compiletest test:false 5.909
2025-03-23T09:58:12.4833880Z [RUSTC-TIMING] compiletest test:false 0.253
2025-03-23T09:58:12.4861940Z     Finished `release` profile [optimized] target(s) in 23.17s
2025-03-23T09:58:12.4931970Z ##[endgroup]
2025-03-23T09:58:12.4933160Z [TIMING] core::build_steps::tool::ToolBuild { compiler: Compiler { stage: 0, host: aarch64-apple-darwin, forced_compiler: false }, target: aarch64-apple-darwin, tool: "compiletest", path: "src/tools/compiletest", mode: ToolStd, source_type: InTree, extra_features: [], allow_features: "test", cargo_args: [], artifact_kind: Binary } -- 23.184
2025-03-23T09:58:12.4934460Z [TIMING] core::build_steps::tool::Compiletest { compiler: Compiler { stage: 0, host: aarch64-apple-darwin, forced_compiler: false }, target: aarch64-apple-darwin } -- 0.000
2025-03-23T09:58:12.4935320Z [TIMING] core::build_steps::test::RemoteCopyLibs { compiler: Compiler { stage: 2, host: aarch64-apple-darwin, forced_compiler: false }, target: aarch64-apple-darwin } -- 0.000
2025-03-23T09:58:13.7376510Z ##[group]Testing stage2 compiletest suite=ui mode=ui (aarch64-apple-darwin)
2025-03-23T09:58:14.6495420Z
2025-03-23T09:58:14.6495780Z thread 'main' panicked at src/tools/compiletest/src/header.rs:1538:17:
2025-03-23T09:58:14.6496710Z missing LLVM component wasm, and COMPILETEST_REQUIRE_ALL_LLVM_COMPONENTS is set
2025-03-23T09:58:14.6497250Z note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
2025-03-23T09:58:14.6543630Z Build completed unsuccessfully in 0:23:36
2025-03-23T09:58:14.6678530Z   local time: Sun Mar 23 09:58:14 UTC 2025
2025-03-23T09:58:14.7855570Z   network time: Sun, 23 Mar 2025 09:58:14 GMT
2025-03-23T09:58:14.7886920Z ##[error]Process completed with exit code 1.
2025-03-23T09:58:14.7998940Z Post job cleanup.
2025-03-23T09:58:15.0894090Z [command]/opt/homebrew/bin/git version
2025-03-23T09:58:15.0969920Z git version 2.49.0
2025-03-23T09:58:15.1010290Z Copying '/Users/runner/.gitconfig' to '/Users/runner/work/_temp/70f16c92-9d9f-4c2f-9d2b-528d5e2266d8/.gitconfig'
2025-03-23T09:58:15.1017410Z Temporarily overriding HOME='/Users/runner/work/_temp/70f16c92-9d9f-4c2f-9d2b-528d5e2266d8' before making global git config changes
2025-03-23T09:58:15.1018120Z Adding repository directory to the temporary git global config as a safe directory
2025-03-23T09:58:15.1021380Z [command]/opt/homebrew/bin/git config --global --add safe.directory /Users/runner/work/rust/rust
2025-03-23T09:58:15.1104750Z [command]/opt/homebrew/bin/git config --local --name-only --get-regexp core\.sshCommand
2025-03-23T09:58:15.1156810Z [command]/opt/homebrew/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2025-03-23T09:58:15.1973790Z Entering 'library/backtrace'
2025-03-23T09:58:15.2121350Z Entering 'library/stdarch'
2025-03-23T09:58:15.2269930Z Entering 'src/doc/book'
2025-03-23T09:58:15.2420260Z Entering 'src/doc/edition-guide'
2025-03-23T09:58:15.2561610Z Entering 'src/doc/embedded-book'
2025-03-23T09:58:15.2702630Z Entering 'src/doc/nomicon'
2025-03-23T09:58:15.2839020Z Entering 'src/doc/reference'
```

You do a rough guess on what the problem is here
```
2025-03-23T09:58:14.6495780Z thread 'main' panicked at src/tools/compiletest/src/header.rs:1538:17:
2025-03-23T09:58:14.6496710Z missing LLVM component wasm, and COMPILETEST_REQUIRE_ALL_LLVM_COMPONENTS is set
2025-03-23T09:58:14.6497250Z note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
but compiletest never says which testfile the problem occurred with which is bad 😅

--- a/src/tools/compiletest/src/header.rs
+++ b/src/tools/compiletest/src/header.rs
@@ -1385,7 +1385,7 @@ pub fn make_test_description<R: Read>(
             decision!(cfg::handle_ignore(config, ln));
             decision!(cfg::handle_only(config, ln));
             decision!(needs::handle_needs(&cache.needs, config, ln));
-            decision!(ignore_llvm(config, ln));
+            decision!(ignore_llvm(config, path, ln));
             decision!(ignore_cdb(config, ln));
             decision!(ignore_gdb(config, ln));
             decision!(ignore_lldb(config, ln));
@@ -1525,7 +1525,7 @@ fn ignore_lldb(config: &Config, line: &str) -> IgnoreDecision {
     IgnoreDecision::Continue
 }
 
-fn ignore_llvm(config: &Config, line: &str) -> IgnoreDecision {
+fn ignore_llvm(config: &Config, path: &Path, line: &str) -> IgnoreDecision {
     if let Some(needed_components) =
         config.parse_name_value_directive(line, "needs-llvm-components")
     {
@@ -1536,8 +1536,9 @@ fn ignore_llvm(config: &Config, line: &str) -> IgnoreDecision {
         {
             if env::var_os("COMPILETEST_REQUIRE_ALL_LLVM_COMPONENTS").is_some() {
                 panic!(
-                    "missing LLVM component {}, and COMPILETEST_REQUIRE_ALL_LLVM_COMPONENTS is set",
-                    missing_component
+                    "missing LLVM component {}, and COMPILETEST_REQUIRE_ALL_LLVM_COMPONENTS is set: {}",
+                    missing_component,
+                    path.display()
                 );
             }
             return IgnoreDecision::Ignore {

